sdk/typescript/src/cryptography/publickey.ts

```diff
@@ -63,6 +66,35 @@ export abstract class PublicKey {
 		return toB64(suiPublicKey);
 	}
 
+	verifyWithIntent(
+		bytes: Uint8Array,
+		signature: Uint8Array,
+		intent: IntentScope,
+	): Promise<boolean> {
+		const intentMessage = messageWithIntent(intent, bytes);
+		const digest = blake2b(intentMessage, { dkLen: 32 });
+
+		return this.verify(digest, signature);
+	}
+
+	/**
+	 * Verifies that the signature is valid for for the provided PersonalMessage
+	 */
+	verifyPersonalMessage(message: Uint8Array, signature: Uint8Array): Promise<boolean> {
```

Comment:
We should check to see what the expectation is for messages from Rust. I realize now that this approach means that if you serialize the rust bytes with the `PersonalMessage` struct wrapping, it'll fail to verify here because we'll re-wrap it. So we should check to see what rust returns you as bytes here.

Reply:
yeah, maybe worth asking @joyqvq about this in our meeting tomorrow

Reply:
So the way this works in Rust (as far as I can tell) is that to create a signature for a PersonalMessage, you need to create a PersonalMessage struct to pass into the Sign method.  There isn't really a message that is "returned" from rust.  If you serialize an PersonalMessage struct you get the bcs bytes, but if you just serialize the message provided to the struct, you get the raw bytes, so that doesn't really provide a clear answer.

My inclination here is that we don't provide a way to construct a PersonalMessage in the TS SDK, so I would assume the bytes provided are the raw message bytes, and we should eventually have a PersonalMessage class or something that you can do PersonalMessage.fromBcs(bytes) or something that could also be passed here, but that would come after doing our bcs rewrite